### PR TITLE
docs(core): Add more ts docs for the core module

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

Add more typescript docs.

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: packages/core/src/application.ts [hunk 1]
```diff
@@ -75,10 +80,13 @@ export class Application extends Context {
...
    */
-  public server<T extends Server>(ctor: Constructor<T>, name?: string) {
+  public server<T extends Server>(
+    ctor: Constructor<T>,
+    name?: string,
+  ): Binding {
     const suffix = name || ctor.name;
     const key = `${CoreBindings.SERVERS}.${suffix}`;
-    this.bind(key)
+    return this.bind(key)
```
Comment (kjdelisle) [inline packages/core/src/application.ts]: This isn't a documentation change, this is an actual code change.

File: packages/core/src/application.ts [hunk 2]
```diff
@@ -103,8 +111,8 @@ export class Application extends Context {
    * @param {Constructor<Server>[]} ctors An array of Server constructors.
    * @memberof Application
    */
-  public servers<T extends Server>(ctors: Constructor<T>[]) {
-    ctors.map(ctor => this.server(ctor));
+  public servers<T extends Server>(ctors: Constructor<T>[]): Binding[] {
+    return ctors.map(ctor => this.server(ctor));
```
Comment (kjdelisle) [inline packages/core/src/application.ts]: Same goes here.

**Comment by raymondfeng** [comment]
@kjdelisle Reverted the code changes. PTAL.

File: packages/core/src/component.ts [hunk 3]
```diff
@@ -4,20 +4,38 @@
...
 
+/**
+ * A map of name/class pairs for binding providers
+ */
 export interface ProviderMap {
   [key: string]: Constructor<Provider<BoundValue>>;
 }
 
+/**
+ * A component declares a set of artifacts so that they be contributed to an
```
Comment (jannyHou) [inline packages/core/src/component.ts]: nitpick:

> they be contributed 

--> "they can be contributed"

**Pull request merged**